PUSH_TO_FILE.md

```diff
@@ -82,201 +292,220 @@ for a description of each environment variable setting:
-        - name: conjur-secrets
-          emptyDir:
-            medium: Memory
+Custom templates support global functions native to Go's `text/template`
+package. The following is an example of using template function to HTML
+escape/encode a secret value.
 
-      imagePullSecrets:
-        - name: dockerpullsecret
 ```
+{{ secret "alias" | html }}
+{{ secret "alias" | urlquery }}
+```
+
+If the value retrieved from Conjur for `alias` is `"<Hello@World!>"`,
+the following file content will be rendered, each HTML escaped and encoded,
+respectively:
+
+```
+&lt;Hello;@World!&gt;
+%3CHello%40World%21%3E
+```
+
+For a full list of global Go text template functions, reference the official
+[`text/template` documentation](https://pkg.go.dev/text/template#hdr-Functions).
+
+### Execution "Double-Pass"
+
+To avoid leaking sensitive secret data to logs, and to ensure that a
+misconfigured Push to File workflow fails fast, Push to File implements a
+"double-pass" execution of custom templates. The template "first-pass" runs
+before secrets are retrieved from Conjur, and validates that the provided custom
+template successfully executes given `"REDACTED"` for each secret value.
+Redacting secret values allows for secure, complete error logging for
+malformed templates. The template "second-pass" runs when rendering secret
+files, and error messages during this stage are sanitized. Custom templates that
+pass the "first-pass" and fail the "second-pass" require experimenting locally
+to identify bugs.
+
+_**NOTE**: Custom templates should not branch conditionally on secret values.
+This may result in a template first-pass execution that doesn't validate all
+branches of the custom template._
+
+### Example Custom Templates: Direct reference to secret values
+
+The following is an example of using a custom template to render secret data by
+referencing secrets directly using the custom template function `secret`.
 
-This will create a file /opt/secrets/conjur/redis.yaml, with contents as below.
 ```
-"api-url": "value-dev/redis/api-url"
-"admin-username": "value-dev/redis/username"
-"admin-password": "value-dev/redis/password"
+conjur.org/secret-file-template.direct-reference: |
+  username | {{ secret "db-username" }}
+  password | {{ secret "db-password" }}
 ```
 
-Below are code snippet is for JSON output.
+Assuming that the following secrets have been retrieved for secret group
+`direct-reference`:
 
 ```
-conjur.org/conjur-secrets.cache: |
-  - dev/redis/api-url
-  - admin-username: dev/redis/username
-  - admin-password: dev/redis/password
-     conjur.org/secret-file-path.cache: "./testdata/redis.json"
-     conjur.org/secret-file-format.cache: "json"
+db-username: admin
+db-password: my$ecretP@ss!
 ```
 
-This will create a file redis.json, with contents as below.
+
 ```
-{"api-url":"value-dev/redis/api-url","admin-username":"value-dev/redis/username","admin-password
-":"value-dev/redis/password"}
+username | admin
```

Comment:
Added this in. For each custom template example, I made sure to include the `conjur.org/secret-file-path` annotation, because it's required as of now, and I didn't want to spark confusion. Added comments before the output snippets:

> Secrets Provider will render the following content for the file `/conjur/secrets/<added-file>`